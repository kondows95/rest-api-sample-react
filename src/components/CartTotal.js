import React from 'react';
import PropTypes from 'prop-types'
import Button from '@material-ui/core/Button'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import CardActions from '@material-ui/core/CardActions'
import Box from '@material-ui/core/Box'
import { Link } from 'react-router-dom'

const CartTotal = ({totalPrice, totalQuantity}) => {
  return (
    <Box display="flex" flexDirection="column">
      <Card>
        <CardContent>
          <Box mt={1} fontSize="h6.fontSize" fontWeight={600}>
            Total {totalPrice} Ks
          </Box>
        </CardContent>
       
        <CardActions >
          <Box style={{ width: '100%' }}>
          <Link to={totalQuantity > 0 ? "/checkout": "#"}  style={{ textDecoration: 'none'  }}>
          <Button fullWidth={true}  disabled={totalQuantity > 0 ? false: true}   variant="contained" color="secondary">
            Confirm Order
          </Button>
          </Link>
          </Box>
        </CardActions>

        <CardContent>
        <Link to="/" style={{ textDecoration: 'none' }}>
          <Button fullWidth={true}  color="primary">
            <Box fontSize={11} fontWeight={300} >Continue Shopping</Box>
          </Button>
          </Link>
        </CardContent>
      </Card>
    </Box>  
  )
}

CartTotal.propTypes = {
  totalPrice: PropTypes.number.isRequired,
}

export default CartTotal
