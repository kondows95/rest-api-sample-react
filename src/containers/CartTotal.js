import { connect } from 'react-redux'
import CartTotal from '../components/CartTotal'

export default connect(
  (state) => ({
    totalPrice: state.cart.totalPrice,
    totalQuantity:  state.cart.totalQuantity,
  })
)(CartTotal)
