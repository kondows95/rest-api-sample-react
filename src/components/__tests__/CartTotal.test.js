import React from 'react'
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import CartTotal from '../CartTotal';
import { BrowserRouter as Router } from 'react-router-dom';

let container
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  document.body.outerHTML = '<body></body>'
  container = null
})

const valueSetter = Object.getOwnPropertyDescriptor(window.HTMLInputElement.prototype, "value").set;
const totalPrice = 2000;
const totalQuantity = 2;
describe("CartTotal component snapshot", () => {

  it('matches the snapshot', () => {
    const CartTotalSnapshot = renderer.create(
      <Router>
        <CartTotal totalPrice={totalPrice} totalQuantity={totalQuantity} />
      </Router>
    ).toJSON();
    expect(CartTotalSnapshot).toMatchSnapshot();
  })
})

describe("testing cart table", () => {
  it('Component render comfirm order link to /checkout and continue shopping link to /', () => {
    act(() => {
      ReactDOM.render((
        <Router>
          <CartTotal totalPrice={totalPrice} totalQuantity={totalQuantity} />
        </Router>
      ), container);
    });
    const linkArr = document.querySelectorAll('a');
    expect(linkArr[0].textContent.toUpperCase()).toBe("CONFIRM ORDER")
    expect(linkArr[0].getAttribute("href")).toBe("/checkout")
    expect(linkArr[1].textContent.toUpperCase()).toBe("CONTINUE SHOPPING")
    expect(linkArr[1].getAttribute("href")).toBe("/")
  });
  it(' comfirm order link to # because totalquantity <= 0', () => {
    act(() => {
      ReactDOM.render((
        <Router>
          <CartTotal totalPrice={0} totalQuantity={0} />
        </Router>
      ), container);
    });
    const linkArr = document.querySelectorAll('a');
    expect(linkArr[0].textContent.toUpperCase()).toBe("CONFIRM ORDER")
    expect(linkArr[0].getAttribute("href")).toBe("/")
  });
})

