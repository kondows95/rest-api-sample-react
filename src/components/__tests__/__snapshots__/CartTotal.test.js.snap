// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartTotal component snapshot matches the snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-2"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
  >
    <div
      className="MuiCardContent-root"
    >
      <div
        className="MuiBox-root MuiBox-root-32"
      >
        Total 
        2000
         Ks
      </div>
    </div>
    <div
      className="MuiCardActions-root MuiCardActions-spacing"
    >
      <div
        className="MuiBox-root MuiBox-root-35"
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <a
          href="/checkout"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-fullWidth"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Confirm Order
            </span>
          </button>
        </a>
      </div>
    </div>
    <div
      className="MuiCardContent-root"
    >
      <a
        href="/"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-fullWidth"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <div
              className="MuiBox-root MuiBox-root-56"
            >
              Continue Shopping
            </div>
          </span>
        </button>
      </a>
    </div>
  </div>
</div>
`;
